### PR TITLE
Add global seed fixture for reproducible tests

### PR DESCRIPTION
**Pull request by MagnusBuehler**
Defines a session-scoped fixture in conftest.py that sets seeds for torch, numpy, and random to ensure deterministic test behavior across sessions.

## Motivation and Context
Multiple tests rely on randomly sampled data using random, numpy, and torch. Without a fixed seed, this can lead to non-deterministic and flaky test results. Adding a global seed fixture improves consistent and reproducible test behavior across runs.

## Public API Changes

-   [x] No Public API changes
-   [ ] Yes, Public API changes (Details below)

---

## How Has This Been Tested?
On a local machine the randomly sampled data across test runs are consistent.
---

## Checklist

-   [x] The changes have been tested locally.
-   [ ] Documentation has been updated (if the public API or usage changes).
-   [ ] A entry has been added to `CHANGELOG.md` (if relevant for users).
-   [x] The code follows the project's style guidelines.
-   [ ] I have considered the impact of these changes on the public API.

---

### CODE REVIEW

**Review by noahho — APPROVED** [review]
I think this is a very positive change, hope I'm not overlooking anything here. Thanks a lot!

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# conftest.py
```
Comment (oscarkey) [inline tests/conftest.py]:
```suggestion
```

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+import random
+
+import numpy as np
+import pytest
+import torch
+
+
+@pytest.fixture(autouse=True, scope="session")
+def set_global_seed():
```
Comment (oscarkey) [inline tests/conftest.py]:
```suggestion
def set_global_seed() -> None:
```

File: tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+from __future__ import annotations
+
+import random
+
+import numpy as np
+import pytest
+import torch
+
+
+@pytest.fixture(autouse=True, scope="session")
```
Comment (oscarkey) [inline tests/conftest.py]: Should the scope instead be "function", so the seeds are set before every test? If I understand correctly, this will just set the seed once per test session, which could still result in unpredictable behaviour if the order of the tests is changed (e.g. due to adding/removing tests, or using pytest-xdist to run them in parallel).

File: tests/conftest.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+def set_global_seed():
+    seed = 42
+    torch.manual_seed(seed)
+    torch.cuda.manual_seed_all(seed)
```
Comment (oscarkey) [inline tests/conftest.py]:
```suggestion
```
I think torch.manual_seed() will set the seed for all devices, see https://discuss.pytorch.org/t/difference-between-torch-manual-seed-and-torch-cuda-manual-seed/13848/5, so we don't need the separate call. Otherwise, we would also need to add one for MPS.

File: tests/conftest.py (hunk 3 above)
Comment (noahho) [reply tests/conftest.py]: ohh yes taht's a very good and subtle point 🙌

File: tests/conftest.py (hunk 1 above)
Comment (MagnusBuehler) [reply tests/conftest.py]: agreed.

File: tests/conftest.py (hunk 2 above)
Comment (MagnusBuehler) [reply tests/conftest.py]: added suggested return type.

File: tests/conftest.py (hunk 4 above)
Comment (MagnusBuehler) [reply tests/conftest.py]: @oscarkey I removed `torch.cuda.manual_seed_all()`

File: tests/conftest.py (hunk 3 above)
Comment (MagnusBuehler) [reply tests/conftest.py]: @oscarkey Good catch. I'll set `scope="function"`, such that execution order etc does not matter. Ruff gives a warning, due to `"function"` being the default value, which I disabled ( noqa: PT003), as I think it makes sense to explicitly still set the scope to `"function"` for clarity.

Comment (oscarkey) [reply tests/conftest.py]: makes sense to me! this rule seems a bit ott

**Review by oscarkey — APPROVED** [review]
